**<div align="center">1.	Load the Data:<br>**
•	Load the CSV file into a Pandas DataFrame.<br>
•	Data file : GDP (nominal) per Capita


In [ ]:
import pandas as pd

In [2]:
pd

<module 'pandas' from 'C:\\Users\\nicka\\anaconda3\\Lib\\site-packages\\pandas\\__init__.py'>

In [3]:
student_data = pd.read_csv("student.csv")

gdp = pd.read_csv("GDP_(nominal)_per_Capita.csv")

**<div align="center">2.	Display the First  and Last 3 Rows:<br>**
•	Use the head() and tail() functions to display the first and last 3 rows of the DataFrame.


In [6]:
gdp.head(3)

Unnamed: 0 Country/Territory UN_Region  IMF_Estimate  IMF_Year  \
0           1            Monaco    Europe             0         0   
1           2     Liechtenstein    Europe             0         0   
2           3        Luxembourg    Europe        132372      2023   

   WorldBank_Estimate  WorldBank_Year  UN_Estimate UN_Year  
0              234316            2021       234317    2021  
1              157755            2020       169260    2021  
2              133590            2021       133745    2021

In [7]:
gdp.tail(3)

Unnamed: 0 Country/Territory UN_Region  IMF_Estimate  IMF_Year  \
220         221      Sierra Leone    Africa           415      2023   
221         222       Afghanistan      Asia           611      2020   
222         223           Burundi    Africa           249      2023   

     WorldBank_Estimate  WorldBank_Year  UN_Estimate UN_Year  
220                 480            2021          505    2021  
221                 369            2021          373    2021  
222                 222            2021          311    2021

**<div align="center">3.	Display the dimensions of the data set:<br>**
•	Use appropriate function to find out number rows and columns in the dataset


In [8]:
gdp.shape

(223, 9)

**<div align="center">4.	Display Information about the DataFrame:<br>**
•	Use the info() function to display concise information about the DataFrame, including the data types and non-null values , record your observations


In [9]:
gdp.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 223 entries, 0 to 222
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Unnamed: 0          223 non-null    int64 
 1   Country/Territory   223 non-null    object
 2   UN_Region           223 non-null    object
 3   IMF_Estimate        223 non-null    int64 
 4   IMF_Year            223 non-null    int64 
 5   WorldBank_Estimate  223 non-null    int64 
 6   WorldBank_Year      223 non-null    int64 
 7   UN_Estimate         223 non-null    int64 
 8   UN_Year             223 non-null    object
dtypes: int64(6), object(3)
memory usage: 15.8+ KB


**<div align="center">5.	Subset Using loc:<br>**
•	Use the loc indexer to create a subset of the DataFrame. Select specific rows and columns based on labels.


In [28]:
gdp.loc[5:8,["Country/Territory","IMF_Estimate","UN_Year"]]

Country/Territory UN_Year
5            Norway    2021
6       Switzerland    2021
7         Singapore    2021
8       Isle of Man       0

In [29]:
# Select a range of rows and consecutive columns using loc
gdp.loc[5:8,"Country/Territory":"UN_Year"]

Country/Territory UN_Region  IMF_Estimate  IMF_Year  WorldBank_Estimate  \
5            Norway    Europe        101103      2023               89154   
6       Switzerland    Europe         98767      2023               91992   
7         Singapore      Asia         91100      2023               72794   
8       Isle of Man    Europe             0         0               87158   

   WorldBank_Year  UN_Estimate UN_Year  
5            2021        89242    2021  
6            2021        93525    2021  
7            2021        66822    2021  
8            2019            0       0

**<div align="center">6.	Subset Using iloc:<br>**
•	Use the iloc indexer to create another subset of the DataFrame. Select specific rows and columns based on integer positions ( you choose any column you like )


In [38]:
gdp.iloc[3:8,5:10]  #why dosent it print as a formatted table when single numbers rather than ranges are used?

WorldBank_Estimate  WorldBank_Year  UN_Estimate UN_Year
3              100172            2021       101109    2021
4              114090            2021       112653    2021
5               89154            2021        89242    2021
6               91992            2021        93525    2021
7               72794            2021        66822    2021


**<div align="center">7.	Conditional Subsetting:<br>**
•	Use conditional statements to create a subset of the DataFrame.<br>
•	Subsetting based on specific conditions for numerical columns:<br>
o	Select rows where the IMF_Estimate is greater than 100.<br>
o	Select rows where the WorldBank_Estimate is less than 50.<br>
o	Select rows where the UN_Estimate is between 75 and 150.<br>


In [47]:
imf100 = gdp[gdp["IMF_Estimate"]>100]
imf100

Unnamed: 0 Country/Territory UN_Region  IMF_Estimate  IMF_Year  \
2             3        Luxembourg    Europe        132372      2023   
3             4           Ireland    Europe        114581      2023   
5             6            Norway    Europe        101103      2023   
6             7       Switzerland    Europe         98767      2023   
7             8         Singapore      Asia         91100      2023   
..          ...               ...       ...           ...       ...   
218         219            Malawi    Africa           496      2023   
219         220       South Sudan    Africa           467      2023   
220         221      Sierra Leone    Africa           415      2023   
221         222       Afghanistan      Asia           611      2020   
222         223           Burundi    Africa           249      2023   

     WorldBank_Estimate  WorldBank_Year  UN_Estimate UN_Year  
2                133590            2021       133745    2021  
3                100172            2021       101109    2021  
5                 89154            2021        89242    2021  
6                 91992            2021        93525    2021  
7                 72794            2021        66822    2021  
..                  ...             ...          ...     ...  
218                 635            2021          613    2021  
219                1072            2015          400    2021  
220                 480            2021          505    2021  
221                 369            2021          373    2021  
222                 222            2021          311    2021  

[197 rows x 9 columns]

In [46]:
wb = gdp[gdp["WorldBank_Estimate"]<50]
wb

Unnamed: 0       Country/Territory UN_Region  IMF_Estimate  IMF_Year  \
30           31  British Virgin Islands  Americas             0         0   
45           46                  Taiwan      Asia         33907      2023   
75           76            Cook Islands   Oceania             0         0   
76           77                Anguilla  Americas             0         0   
84           85              Montserrat  Americas             0         0   
195         196                Zanzibar    Africa             0         0   
211         212             North Korea      Asia             0         0   

     WorldBank_Estimate  WorldBank_Year  UN_Estimate UN_Year  
30                    0               0        49444    2021  
45                    0               0            0       0  
75                    0               0        19264    2021  
76                    0               0        19216    2021  
84                    0               0        16199    2021  
195                   0               0         1211    2021  
211                   0               0          654    2021

In [58]:
#Select rows where the UN_Estimate is between 75 and 150.
unest = gdp[(gdp["UN_Estimate"]>75) & (gdp["UN_Estimate"]<150)]
unest        
#No results - checked in csv file

Empty DataFrame
Columns: [Unnamed: 0, Country/Territory, UN_Region, IMF_Estimate, IMF_Year, WorldBank_Estimate, WorldBank_Year, UN_Estimate, UN_Year]
Index: []

**<div align="center">8.	Summary Statistics:<br>**
•	Generate summary statistics for numerical columns (IMF_Estimate, WorldBank_Estimate, UN_Estimate).


In [63]:
gdp.describe()

Unnamed: 0   IMF_Estimate     IMF_Year  WorldBank_Estimate  \
count  223.000000     223.000000   223.000000          223.000000   
mean   112.000000   15351.632287  1787.098655        18927.417040   
std     64.518731   22550.899445   650.695912        29103.564915   
min      1.000000       0.000000     0.000000            0.000000   
25%     56.500000    1406.500000  2023.000000         2273.500000   
50%    112.000000    5421.000000  2023.000000         6805.000000   
75%    167.500000   19697.000000  2023.000000        23715.000000   
max    223.000000  132372.000000  2023.000000       234316.000000   

       WorldBank_Year    UN_Estimate  
count      223.000000     223.000000  
mean      1957.278027   17767.304933  
std        353.145867   28698.104167  
min          0.000000       0.000000  
25%       2021.000000    2039.000000  
50%       2021.000000    6396.000000  
75%       2021.000000   20740.000000  
max       2021.000000  234317.000000

**<div align="center">9.	Analyse based on nonnumeric column:<br>**
•	Count the number of countries in each UN region.<br>
•	Create subsets for the following UN regions<br>
o	Africa – Call this subset GDP_Africa<br>
o	Europe – Call this subset GDP_Europe<br>
o	Create summary statistics for the above two regions and compare. Write short paragraphs to describe your findings. <br>
Feel free to make any assumptions. 


In [75]:
# number of countries in each UN regions - looked up and used https://stackoverflow.com/questions/39607540/count-the-number-of-occurrence-of-values-based-on-another-column
# could also use: GPD_Africa.count() after creating subsets
gdp.groupby("UN_Region").size().sort_values(ascending=False) \
  .reset_index(name="Country/Territory")

UN_Region  Country/Territory
0    Africa                 55
1      Asia                 51
2  Americas                 48
3    Europe                 48
4   Oceania                 20
5     World                  1

In [ ]:
GDP_Africa = gdp[gdp["UN_Region"]=="Africa"]
GDP_Europe = gdp[gdp["UN_Region"]=="Europe"]

In [76]:
GDP_Africa.describe()

Unnamed: 0  IMF_Estimate     IMF_Year  WorldBank_Estimate  \
count   55.000000     55.000000    55.000000           55.000000   
mean   178.818182   2802.345455  1986.218182         2470.836364   
std     35.501743   3421.329481   272.781264         2772.447680   
min     74.000000      0.000000     0.000000            0.000000   
25%    156.500000    833.000000  2023.000000          762.000000   
50%    188.000000   1475.000000  2023.000000         1319.000000   
75%    206.500000   3696.000000  2023.000000         3221.500000   
max    223.000000  19536.000000  2023.000000        14653.000000   

       WorldBank_Year   UN_Estimate  
count       55.000000     55.000000  
mean      1983.963636   2417.927273  
std        272.476092   2521.291930  
min          0.000000    311.000000  
25%       2021.000000    779.000000  
50%       2021.000000   1361.000000  
75%       2021.000000   3320.500000  
max       2021.000000  12085.000000

In [77]:
GDP_Europe.describe()

Unnamed: 0   IMF_Estimate     IMF_Year  WorldBank_Estimate  \
count   48.000000      48.000000    48.000000           48.000000   
mean    52.687500   34446.750000  1812.270833        45193.687500   
std     40.319024   31085.656917   624.518870        43984.130016   
min      1.000000       0.000000     0.000000         4836.000000   
25%     19.500000   10192.500000  2023.000000        16978.250000   
50%     46.000000   27053.000000  2023.000000        31795.500000   
75%     75.750000   51774.500000  2023.000000        58583.250000   
max    143.000000  132372.000000  2023.000000       234316.000000   

       WorldBank_Year    UN_Estimate  
count       48.000000      48.000000  
mean      2020.625000   40610.791667  
std          2.038199   45233.986703  
min       2007.000000       0.000000  
25%       2021.000000   11468.250000  
50%       2021.000000   27400.000000  
75%       2021.000000   51800.250000  
max       2021.000000  234317.000000

Africa contains the most countries with 55, followed by Asia with 51, the Americas and Europe both with 48 and Oceania with 20. The data contained IMF, World Bank and UN estimates of GDP across these countries.  IMF data between 2020 - 2023 (with 26 empty records), World bank data from 2007 - 2021 with 7 empty records  while the UN data was taken bin 2021 with 8 empty records. Given the range of time that the data was taken, the World Bank data has validity issues as it wouldnt take into account potential economic growth for data collected at the start of the period. 

According to the IMF data, African countries had a mean GDP of 2802.35 compared to 34446.75 in Europe. The World Bank estimated African GDP at 2470.84 and European GDP at 45193.69, while the UN estimated African GDP at 2417.92 compared to 40610.79 in Europe. By all measures a large disparity was found between Europe and African with African countries having much lower GDPs than those in Europe. 

By all measurues, the standard deviation for gdp was much greater in Europe (IMF:31085, WB:43984, UN:45234) compared to Africa (IMF:3421, WB:2772, UN:2521). This indicates a larger disparity in gdp in Europe perhaps due to higher GPD in western European countries compared to the east. The lower disparity in Africa indicates a generally low GDP across the continent.





